packages/react-core/src/components/Menu/examples/Menu.md

```diff
@@ -692,3 +700,535 @@ class MenuOptionMultiSelect extends React.Component {
   }
 }
 ```
+
+### With drilldown
+
+```js
+import React from 'react';
+import { Menu, MenuContent, MenuList, MenuItem, Divider, DrilldownMenu } from '@patternfly/react-core';
+import StorageDomainIcon from '@patternfly/react-icons/dist/js/icons/storage-domain-icon';
+import CodeBranchIcon from '@patternfly/react-icons/dist/js/icons/code-branch-icon';
+import LayerGroupIcon from '@patternfly/react-icons/dist/js/icons/layer-group-icon';
+import CubeIcon from '@patternfly/react-icons/dist/js/icons/cube-icon';
+import {
+  MenuItemStartRollout,
+  MenuItemPauseRollout,
+  MenuItemStorage,
+  MenuItemAppGrouping,
+  MenuItemLabels
+} from './Submenus.js';
+
+class MenuWithDrilldown extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      activeItem: null,
+      menuDrilledIn: [],
+      drilldownPath: [],
+      menuHeights: {},
+      activeMenu: 'rootMenu'
+    };
+    this.onSelect = (event, itemId) => {
+      if (itemId.startsWith('group:')) {
+        console.log(`selected sub-menu: ${itemId.split('group:')[1]}`);
+      } else {
+        console.log(`selected: ${itemId}`);
+        this.setState({
+          activeItem: itemId
+        });
+      }
+    };
+    this.drillIn = (fromMenuId, toMenuId, pathId) => {
+      this.setState({
+        menuDrilledIn: [...this.state.menuDrilledIn, fromMenuId],
+        drilldownPath: [...this.state.drilldownPath, pathId],
+        activeMenu: toMenuId
+      });
+    };
+    this.drillOut = toMenuId => {
+      const menuDrilledInSansLast = this.state.menuDrilledIn.slice(0, this.state.menuDrilledIn.length - 1);
+      const pathSansLast = this.state.drilldownPath.slice(0, this.state.drilldownPath.length - 1);
+      this.setState({
+        menuDrilledIn: menuDrilledInSansLast,
+        drilldownPath: pathSansLast,
+        activeMenu: toMenuId
+      });
+    };
+    this.setHeight = (menuId, height) => {
+      if (!this.state.menuHeights[menuId]) {
+        this.setState({
+          menuHeights: {
+            ...this.state.menuHeights,
+            [menuId]: height
+          }
+        });
+      }
+    };
+  }
+
+  render() {
+    const { activeItem, menuDrilledIn, drilldownPath, activeMenu, menuHeights } = this.state;
+
+    // Menus
+    const rootMenu = () => (
+      <Menu
+        id="rootMenu"
+        containsDrilldown
+        isMenuDrilledIn={menuDrilledIn.includes('rootMenu')}
+        onSelect={this.onSelect}
+      >
+        <MenuContent
+          getHeight={height => this.setHeight('rootMenu', height)}
+          style={{ '--pf-c-menu__content--Height': `${menuHeights[activeMenu]}px` }}
```

Comment:
If we want setting the menu height based off the drilled down MenuItem to be the default behavior can we store that state in MenuContent instead?